operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,324 @@
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config that is persisted to a config map
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+// +kubebuilder:validation:Enum="NoQueueName;QueueName"
+type ManageJobsWithoutQueueNameOption string
+
+const (
+	// NoQueueName means that all jobs will be gated by Kueue
+	NoQueueName ManageJobsWithoutQueueNameOption = "NoQueueName"
+	// QueueName means that the jobs require a queue label.
+	QueueName ManageJobsWithoutQueueNameOption = "QueueName"
+)
+
+// +kubebuilder:validation:Enum="Enabled;Disabled"
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// waitForPodsReady configures gang admission
+	// +optional
+	WaitForPodsReady WaitForPodsReady `json:"waitForPodsReady,omitempty"`
+	// integrations are the types of integrations Kueue will manager
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// featureGates are advanced features for Kueue
+	// +optional
+	FeatureGates map[string]EnabledOrDisabled `json:"featureGates,omitempty"`
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources Resources `json:"resources,omitempty"`
+	// manageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the annotation kueue.x-k8s.io/queue-name.
+	// Allowed values are NoQueueName and QueueName
+	// Default will be QueueName
+	// +kubebuilder:default=QueueName
+	// +optional
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// managedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// Only valid if ManagedJobsWithoutQueueName is NoQueueName
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+	// fairSharing controls the fair sharing semantics across the cluster.
+	// +optional
+	FairSharing FairSharing `json:"fairSharing,omitempty"`
+	// metrics allows one to change if metrics
+	// are enabled or disabled.
+	// Microshift does not enable metrics by default
+	// Default will assume metrics are enabled.
+	// +kubebuilder:default=Enabled
```

Comment:
Would you possibly want to change this default over time? In general for config APIs like this we recommend to set the defaults "in the controller" and not in the API like this. It allows them to be changed in the future if we change our best reccomendation

Reply:
I don't think so.

Microshift requires this be disabled because they don't have monitoring AFAIK.

For all other cases, I'd expect this is to be enabled.

Reply:
I can not add the defaults if you think that is best. I thought it was best practice for APIs.

Reply:
Adding defaults is best practice for workload APIs, but not configuration APIs, and I think this is a config API. Check out https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#defaulting for more detail